tests/population/test_base_population.py

```diff
@@ -283,41 +284,126 @@ def test__assign_demography_with_age_bounds_error(base_simulants, include_sex):
-    generate_population_mock.return_value = full_simulants.drop(columns=["tracked"])
-
-    base_pop = bp.BasePopulation()
-    components = [base_pop]
+def test_scaled_population(config, base_plugins, mocker, fuzzy_checker: FuzzyChecker):
     config.update(
         {
             "population": {
-                "population_size": start_population_size,
-                "include_sex": include_sex,
+                "population_size": 1_000_000,
+                "include_sex": "Both",
             },
             "time": {"step_size": 1},
         },
         layer="override",
     )
-    sim1 = InteractiveContext(
-        components=components, configuration=config, plugin_configuration=base_plugins
-    )
-    unscaled = sim1._data.load("population.structure")
 
-    # Get scaled input data and create a scaled population
-    scalar_data = unscaled.copy().reset_index()
-    scalar_data["scalar"] = 1 - (scalar_data["index"] / len(scalar_data))
-    scalar_data["value"] = scalar_data["value"] * scalar_data["scalar"]
-    scalar_data.drop(columns=["scalar", "index", "age"], inplace=True)
+    # Simple pop data
+    pop_structure = simple_pop_structure()
+    # Simple scalar data to pass to ScaledPopulation
+    scalar_data = simple_pop_structure().drop(columns=["location"])
+    scalar_values = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0]
+    scalar_data["value"] = scalar_values
+
+    # Add data to artifact and mock return for plugin
+    mock_art = MockArtifact()
+    mock_art.write("population.structure", pop_structure)
+    mocker.patch(
+        "tests.mock_artifact.MockArtifactManager._load_artifact",
+    ).return_value = mock_art
+
     scaled_pop = bp.ScaledPopulation(scalar_data)
-    sim2 = InteractiveContext(
+    sim = InteractiveContext(
         components=[scaled_pop], configuration=config, plugin_configuration=base_plugins
     )
-    scaled = sim2._data.load("population.structure")
+    pop = sim.get_population()
+    # Use FuzzyChecker to compare population structure to demographic proportion by
+    # iterating through each age_group/sex combination
+    scaled_structure = pop_structure.copy()
+    scaled_structure["value"] = scaled_structure["value"] * scalar_data["value"]
+    for row in range(len(scaled_structure)):
+        row_data = scaled_structure.iloc[row]
+        age_start = row_data["age_start"]
+        sex = row_data["sex"]
+        # Get proportion of each age group
+        target_proportion = (
+            row_data["value"]
+            / scaled_structure.loc[scaled_structure["sex"] == sex]["value"].sum()
+        )
+        number_of_sims = len(
+            pop.loc[
+                (pop["age"] >= age_start)
+                & (pop["age"] <= row_data["age_end"])
+                & (pop["sex"] == sex)
+            ]
+        )
+        fuzzy_checker.fuzzy_assert_proportion(
+            observed_numerator=number_of_sims,
+            observed_denominator=len(pop.loc[pop["sex"] == sex]),
```

Comment:
I would have expected the observed denominator to be `len(pop)`, and the target proportion to be `row_data["value"] / pop["value"].sum()`. That makes this test much simpler and should be equivalent?

Reply:
Updated the implementation of this.